src/streaming/text/TextSourceBuffer.js

```diff
@@ -151,7 +151,6 @@ function TextSourceBuffer(config) {
         }
 
         textTrackInfo.labels = mediaInfo.labels;
-        textTrackInfo.id = mediaInfo.id ? mediaInfo.id : mediaInfo.index; // AdaptationSet id (an unsigned int) as it's optional parameter, use mediaInfo.index
```

Comment:
shouldnt we keep the id part and just remove index?

Reply:
id already set in MediaInfo